### PR TITLE
LPS-115548 BreadCrumb Contributor

### PR DESCRIPTION
**Pull request by AliciaGarciaGarcia**

### CODE REVIEW

File: modules/apps/site-navigation/site-navigation-taglib/src/main/java/com/liferay/site/navigation/taglib/servlet/taglib/BreadcrumbTag.java [hunk 1]
```diff
@@ -58,13 +66,34 @@ protected String getPage() {
 
 	@Override
 	protected void setAttributes(HttpServletRequest httpServletRequest) {
+		List<BreadcrumbEntry> breadcrumbEntries = _breadcrumbEntries;
+
+		try {
+			_serviceTracker = ServiceTrackerFactory.open(
```
Comment (AliciaGarciaGarcia) [inline modules/apps/site-navigation/site-navigation-taglib/src/main/java/com/liferay/site/navigation/taglib/servlet/taglib/BreadcrumbTag.java:72]: I had to use this cause
 SF : `Do not use com.liferay.registry classes in modules, see LPS-62989: ./modules/apps/site-navigation/site-navigation-taglib/src/main/java/com/liferay/site/navigation/taglib/servlet/taglib/BreadcrumbTag.java (SourceCheck:JavaModuleIllegalImportsCheck)`
I tried other methods, but this is the simplest I found

File: modules/apps/asset/asset-list-web/src/main/resources/META-INF/resources/view.jsp [hunk 2]
```diff
@@ -34,6 +34,10 @@ AssetListManagementToolbarDisplayContext assetListManagementToolbarDisplayContex
 </portlet:actionURL>
 
 <aui:form action="<%= deleteAssetListEntryURL %>" cssClass="container-fluid-1280" name="fm">
+	<liferay-ui:breadcrumb
```
Comment (AliciaGarciaGarcia) [inline modules/apps/asset/asset-list-web/src/main/resources/META-INF/resources/view.jsp:37]: With this, you add Asset Library > asset name

It lacks of the
Asset Library > asset name > _Home_

**Comment by 4lejandrito** [comment]
I am reviewing this.

**Comment by 4lejandrito** [comment]
See https://github.com/liferay-lima/liferay-portal/pull/508.

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:forward

**Comment by austinchiang** [comment]
Need to wait for https://github.com/liferay-frontend/liferay-portal/pull/173 to be merged for tests to pass.

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by austinchiang** [comment]
**Reviewing...** :hourglass_flowing_sand: :sunglasses: :headphones:

**Comment by austinchiang** [comment]
Since that bug might take some time to fix I've gone ahead and disabled the tests and resent the pull [here](https://github.com/liferay-lima/liferay-portal/pull/521)

**Pull request closed without merging**